pyomo/contrib/mindtpy/tests/test_mindtpy_feas_pump.py

```diff
@@ -0,0 +1,292 @@
+# -*- coding: utf-8 -*-
+"""Tests for the MindtPy solver."""
+from math import fabs
+import pyomo.core.base.symbolic
+from pyomo.core.expr import template_expr
+import pyutilib.th as unittest
+from pyomo.contrib.mindtpy.tests.eight_process_problem import \
+    EightProcessFlowsheet
+from pyomo.contrib.mindtpy.tests.MINLP_simple import SimpleMINLP as SimpleMINLP
+from pyomo.contrib.mindtpy.tests.MINLP2_simple import SimpleMINLP as SimpleMINLP2
+from pyomo.contrib.mindtpy.tests.MINLP3_simple import SimpleMINLP as SimpleMINLP3
+from pyomo.contrib.mindtpy.tests.from_proposal import ProposalModel
+from pyomo.contrib.mindtpy.tests.constraint_qualification_example import ConstraintQualificationExample
+from pyomo.contrib.mindtpy.tests.online_doc_example import OnlineDocExample
+from pyomo.environ import SolverFactory, value
+from pyomo.environ import *
+from pyomo.solvers.tests.models.LP_unbounded import LP_unbounded
+from pyomo.solvers.tests.models.QCP_simple import QCP_simple
+from pyomo.solvers.tests.models.MIQCP_simple import MIQCP_simple
+from pyomo.opt import TerminationCondition
+from pyomo.contrib.gdpopt.util import is_feasible
+from pyomo.util.infeasible import log_infeasible_constraints
+from pyomo.contrib.mindtpy.tests.feasibility_pump1 import Feasibility_Pump1
+from pyomo.contrib.mindtpy.tests.feasibility_pump2 import Feasibility_Pump2
+
+required_solvers = ('ipopt', 'cplex')
```

Comment:
The feaspump tests should run with glpk, no necessarily with cplex

Reply:
I set cplex here because there is one test includes `fp_master_norm='L2'`. GLPK is not capable of solving MIQP.

Reply:
GLPK should be the main testing for this functionality and we should have CPLEX only for the L2 norm problem. Besides, a warning should be raised when the MIP solver cannot handle quadratic objectives.